src/test/java/io/stargate/sgv2/jsonapi/service/embedding/operation/VectorizeDataTest.java

```diff
@@ -0,0 +1,169 @@
+      Throwable failure = catchThrowable(() -> vectorizeData.vectorize(documents));
+      assertThat(failure)
+          .isInstanceOf(JsonApiException.class)
+          .hasFieldOrPropertyWithValue("errorCode", ErrorCode.SHRED_BAD_VECTORIZE_VALUE)
+          .hasFieldOrPropertyWithValue("message", "$vectorize search needs to be text value");
+    }
+
+    @Test
+    public void testNullValues() {
+      List<JsonNode> documents = new ArrayList<>();
+      for (int i = 0; i < 2; i++) {
+        documents.add(objectMapper.createObjectNode().put("$vectorize", (String) null));
+      }
+
+      VectorizeData vectorizeData = new VectorizeData(testService, objectMapper.getNodeFactory());
+      vectorizeData.vectorize(documents);
+      for (JsonNode document : documents) {
+        assertThat(document.has("$vectorize")).isTrue();
+        assertThat(document.has("$vector")).isTrue();
+      }
+    }
+  }
+
+  @Nested
+  public class SortClauseValues {
+    @Test
+    public void sortClauseValues() {
+      SortExpression sortExpression = SortExpression.vectorizeSearch("test data");
+      List<SortExpression> sortExpressions = new ArrayList<>();
+      sortExpressions.add(sortExpression);
+      SortClause sortClause = new SortClause(sortExpressions);
+      VectorizeData vectorizeData = new VectorizeData(testService, objectMapper.getNodeFactory());
+      vectorizeData.vectorize(sortClause);
+      assertThat(sortClause.hasVsearchClause()).isTrue();
+      assertThat(sortClause.hasVectorizeSearchClause()).isFalse();
+      assertThat(sortClause.sortExpressions().get(0).vector()).isNotNull();
+      assertThat(sortClause.sortExpressions().get(0).vector().length).isEqualTo(3);
+    }
+  }
+
+  @Nested
+  public class UpdateClauseValues {
+    @Test
+    public void updateClauseSetValues() throws Exception {
+      String json =
+          """
+        {
+          "findOneAndUpdate": {
+            "filter" : {"_id" : "id"},
+            "update" : {"$set" : {"$vectorize" : "New York"}}
+          }
+        }
+        """;
+      FindOneAndUpdateCommand command = objectMapper.readValue(json, FindOneAndUpdateCommand.class);
+      UpdateClause updateClause = command.updateClause();
+      VectorizeData vectorizeData = new VectorizeData(testService, objectMapper.getNodeFactory());
+      vectorizeData.vectorizeUpdateClause(updateClause);
+      final ObjectNode setNode = updateClause.updateOperationDefs().get(UpdateOperator.SET);
+      assertThat(setNode.has("$vectorize")).isTrue();
+      assertThat(setNode.has("$vector")).isTrue();
+      assertThat(setNode.get("$vector").isArray()).isTrue();
+      assertThat(setNode.get("$vector").size()).isEqualTo(3);
+    }
+
+    @Test
+    public void updateClauseSetOnInsertValues() throws Exception {
+      String json =
+          """
+        {
+          "findOneAndUpdate": {
+            "filter" : {"_id" : "id"},
+            "update" : {"$setOnInsert" : {"$vectorize" : "New York"}}
```

Comment:
what is the difference between this $setOnInsert and upper $set

Reply:
When upsert option is set to true during update operation, we will create new document if no document found based on condition. setOnInsert will work when this a new document.

Reply:
`$setOnInsert` is only applied in case a new document is inserted, but not if one exists.
Used for things like adding id field(s), or creation-time.